portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -49,6 +49,12 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       comments.add(body);
     }
 
+    int numComments = getNumCommentsToDisplay(request);
+
+    if (numComments >= 0 && numComments <= comments.size()) {
+      comments = comments.subList(0, numComments);
+    }
```

Comment:
The `PreparedQuery` has an `asList` method [1] that takes `FetchOptions` [2] object. You can use the the fetch options to specify a limit so that you don't get more results back than you need.

[1] https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery.html#asList-com.google.appengine.api.datastore.FetchOptions-
[2] https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/FetchOptions.html